### PR TITLE
Add support for facet typeahead to OpenAPI spec

### PR DESCRIPTION
**Pull request by janmichaelyu**
See #8

### CODE REVIEW

**Comment by patrickmcelwee** [comment]
Thanks for this!!

Can you squash these commits into one? (try `git rebase -i`, which has some instructions on how to do it). If you force-push the squashed commit to your fork, it will automatically update this PR. This should get rid of the merge commit and make the git history cleaner.

@grtjn says he can help walk you through this git stuff, if that's helpful.

**Comment by janmichaelyu** [comment]
@patrickmcelwee how does this look?

File: spec/search-api.json [hunk 1]
```diff
@@ -163,6 +213,32 @@
...
       },
+      "SuggestRequest": {
+        "type": "object",
+        "properties": {
+          "filters": {
+            "$ref": "#/components/schemas/CombinedFilter"
+          },
+          "options": {
+            "type": "object",
+            "properties": {
```
Comment (grtjn) [inline spec/search-api.json:224]: It looks like these request options are copied from search request, but start and pageLength are not relevant for suggestions. Instead, a `limit` option might be more relevant..

**Comment by grtjn** [comment]
Merging this so that I can apply cleanup more easily..

**Pull request merged**